### PR TITLE
chore: replace MUI components from app menu (DHIS2-9699)

### PR DESCRIPTION
**Pull request by turban**
Partly fixes: https://jira.dhis2.org/browse/DHIS2-9699

This PR removes Material UI components from the app menu. Than plan was to replace with @dhis2/ui components, but I experienced some limitations:
- There is no @dhis2/ui MenuButton component (a button without an outline). I copied the MenuButton from DV: https://github.com/dhis2/data-visualizer-app/tree/master/packages/app/src/components/MenuButton
- I tried the @dhis2/ui DropdownButton for the "Add layer" component, but it was not possible to close the dropdown programatically when a layer was selected. The fallback was to reuse the above MenuButton with a similar look as today.

After this PR: 

Screenshot 2020-10-22 at 19 24 51

Before this PR: 

Screenshot 2020-10-22 at 19 25 10

I still have an issue with popover placement (which I think is caused by conflicting MUI styles), so this PR will only be merged to the `chore/dhis2/ui` branch. 

Screenshot 2020-10-22 at 19 34 15

### CODE REVIEW

File: src/components/layers/layers/AddLayerButton.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const AddLayerButton = () => {
+    const [isOpen, setIsOpen] = useState(false);
+    const [buttonRef, setButtonRef] = useState();
+    const toggleDialog = () => setIsOpen(!isOpen);
+
+    return (
+        <Fragment>
+            <div className={styles.addLayerBtn} ref={setButtonRef}>
+                <MenuButton
+                    secondary
```
Comment (edoardo) [inline src/components/layers/layers/AddLayerButton.js:17]: Left over props?

File: src/components/layers/layers/AddLayerButton.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import styles from './styles/AddLayerButton.module.css';
+
+const AddLayerButton = () => {
+    const [isOpen, setIsOpen] = useState(false);
+    const [buttonRef, setButtonRef] = useState();
+    const toggleDialog = () => setIsOpen(!isOpen);
+
+    return (
+        <Fragment>
+            <div className={styles.addLayerBtn} ref={setButtonRef}>
```
Comment (edoardo) [inline src/components/layers/layers/AddLayerButton.js:15]: I'm not sure about this?
Shouldn't it just use the `useRef` hook?

File: src/components/layers/layers/AddLayerPopover.js [hunk 3]
```diff
@@ -53,24 +42,18 @@ const AddLayerPopover = ({
 
 AddLayerPopover.propTypes = {
     anchorEl: PropTypes.instanceOf(Element),
-    layersDialogOpen: PropTypes.bool,
+    open: PropTypes.bool.isRequired,
```
Comment (edoardo) [inline src/components/layers/layers/AddLayerPopover.js:45]: This prop is not used?

File: src/components/layers/layers/AddLayerPopover.js [hunk 4]
```diff
@@ -53,24 +42,18 @@ const AddLayerPopover = ({
...
     layers: PropTypes.array,
     isSplitView: PropTypes.bool,
     addLayer: PropTypes.func.isRequired,
     editLayer: PropTypes.func.isRequired,
-    closeLayersDialog: PropTypes.func.isRequired,
```
Comment (edoardo) [inline src/components/layers/layers/AddLayerPopover.js:61]: `onClose` missing?

File: src/components/interpretations/styles/InterpretationsToggle.module.css [hunk 5]
```diff
@@ -0,0 +1,10 @@
+.button {
+    position: absolute;
+    right: 0;
+    top: 1px;
+}
+
+.icon {
+    margin-right: 8px;
```
Comment (edoardo) [inline src/components/interpretations/styles/InterpretationsToggle.module.css:8]: Use `spacers` variables?

**Review by edoardo — CHANGES_REQUESTED** [review]
Looks good!
A few comments mostly about some cleaning up of outdated code and props.

**Review by edoardo — APPROVED** [review]

File: src/components/layers/layers/AddLayerPopover.js (hunk 4 above)
Comment (turban) [reply src/components/layers/layers/AddLayerPopover.js:61]: onClose is there :-)

File: src/components/layers/layers/AddLayerButton.js (hunk 1 above)
Comment (turban) [reply src/components/layers/layers/AddLayerButton.js:17]: Well spotted. Now removed.

File: src/components/layers/layers/AddLayerButton.js (hunk 2 above)
Comment (turban) [reply src/components/layers/layers/AddLayerButton.js]: Switched to useRef hook.

File: src/components/layers/layers/AddLayerPopover.js (hunk 3 above)
Comment (turban) [reply src/components/layers/layers/AddLayerPopover.js]: open prop is now removed.

File: src/components/interpretations/styles/InterpretationsToggle.module.css (hunk 5 above)
Comment (turban) [reply src/components/interpretations/styles/InterpretationsToggle.module.css]: Fixed.

File: src/components/layers/layers/AddLayerPopover.js (hunk 4 above)
Comment (turban) [reply src/components/layers/layers/AddLayerPopover.js:61]: onClose is there :-)

**Review by cooper-joe — APPROVED** [review]
LGTM. (ToolbarButton is a component that will be added to `ui`, [pending ticket here](https://jira.dhis2.org/browse/TECH-115). So in the future we should be able to use `ui` here.)

**Pull request merged**